parser/lexer.go

```diff
@@ -222,6 +222,20 @@ func startWithXx(s *Scanner) (tok int, pos Pos, lit string) {
 	return
 }
 
+func startWithNn(s *Scanner) (tok int, pos Pos, lit string) {
+	tok, pos, lit = scanIdentifier(s)
+	// The National Character Set, N'some text' or n'some test'.
+	// See https://dev.mysql.com/doc/refman/5.7/en/string-literals.html
+	// and https://dev.mysql.com/doc/refman/5.7/en/charset-national.html
+	if lit == "N" || lit == "n" {
+		if s.r.peek() == '\'' {
```

Comment:
Normally, `N` is parsed to identifier, but when `N` is followed by `"` without whitespace, try to parse it to underscoreCS